lib/ret/api/token_module.ex

```diff
@@ -0,0 +1,111 @@
+defmodule Ret.Api.TokenModule do
+  @moduledoc """
+  This module should not be used directly.
+
+  It is intended to be used by Guardian.
+  """
+  alias Ret.{Account, Repo}
+  alias Ret.Api.Credentials
+  @behaviour Guardian.Token
+
+  @doc """
+  No concept of validating signature so we just decode the token
+  """
+  def peek(mod, token) do
+    case decode_token(mod, token) do
+      {:ok, %Credentials{} = credentials} -> %{claims: credentials}
+      {:ok, {:error, _reason}} -> nil
+      _ -> nil
+    end
+  end
+
+  @doc """
+  Do not need to generate a token_id here
+  """
+  def token_id, do: nil
+
+  @doc """
+  Builds the default claims for API tokens.
+  """
+  def build_claims(_mod, _resource, _sub, claims \\ %{}, _options \\ []) do
+    {:ok, claims}
+  end
+
+  defp ensure_atom(x) when is_atom(x), do: x
+  defp ensure_atom(x) when is_binary(x), do: String.to_atom(x)
+
+  defp get_account(id) when is_nil(id) do
+    {:ok, nil}
+  end
+
+  defp get_account(id) do
+    case Account.query()
+         |> Account.where_account_id_is(id)
+         |> Repo.one() do
+      nil -> {:error, "Could not find account"}
+      account -> {:ok, account}
+    end
+  end
+
+  @doc """
+  Create a token.
+  """
+  def create_token(_mod, claims, _options \\ []) do
+    token = SecureRandom.urlsafe_base64()
```

Comment:
Perhaps tokens should be prefixed with the credential's sid, so that they can be matched up later?

Reply:
Also, not sure if this is a quirk on my machine but I noticed that `urlsafe_base64` with the default 16 bytes always generates a string with "09" at the end. Using 18 bytes fixes this.

Reply:
RE: Prefix
Good idea. I've added this in https://github.com/mozilla/reticulum/pull/399/commits/50f7df661526dcb68b24fa2913cfb787c6f48a04 . Do you have an opinion on how we should differentiate the `sid` from the rest of the token (to make it easier to read)? I chose `"_"` (e.g. `"XAxwsZ6_VU56NzgzZmRXcmMvdkdkRmhOWWNlbDgz"`). 

RE: 09
Ha, I didn't notice this. Fwiw this is not evidence of security being compromised. See https://github.com/patricksrobertson/secure_random.ex/issues/11 Since we're "paying" for the extra characters anyway, we might as well get that little bit of added entropy (and preempt having to answer this question anytime a user would have discovered the same pattern). I've changed it to use 18 bytes in https://github.com/mozilla/reticulum/pull/399/commits/832898bfef040e02496ffa1178c935a2e80c02eb .

Reply:
It looks like SecureRandom may generate `"_"` as one of its random characters, so I would avoid using that. I'd suggest `"."` since that's not used in any flavor of base64.

Reply:
SecureRandom uses `Base.url_encode64` which uses this set of characters: https://hexdocs.pm/elixir/Base.html#module-base-64-url-and-filename-safe-alphabet